fields/evaluate.go

```diff
@@ -21,113 +21,190 @@ import (
 )
 
 const (
-	AggregateDefault AggregateMethod = iota
-	AggregateCount
-	AggregateMin
-	AggregateMax
-	AggregateSum
-	AggregateAvg
-	AggregateStateHistSeconds
-	AggregateStateHistPercent
-	AggregateStateHistRate
+	TimeAggregationDefault TimeAggregationMethod = iota
+	TimeAggregationCount
+	TimeAggregationMin
+	TimeAggregationMax
+	TimeAggregationSum
+	TimeAggregationAvg
+	TimeAggregationSeconds
+	TimeAggregationPercent
+	TimeAggregationRate
 )
 
-type AggregateMethod uint8
+type TimeAggregationMethod uint8
+
+const (
+	GroupAggregationDefault GroupAggregation = iota
+	GroupAggregationCount
+	GroupAggregationMin
+	GroupAggregationMax
+	GroupAggregationSum
+	GroupAggregationAvg
+)
+
+type GroupAggregation uint8
 
 var (
-	_ encoding.TextMarshaler   = AggregateMethod(0)
-	_ encoding.TextUnmarshaler = (*AggregateMethod)(nil)
+	_ encoding.TextMarshaler   = TimeAggregationMethod(0)
+	_ encoding.TextUnmarshaler = (*TimeAggregationMethod)(nil)
 )
 
-func (m AggregateMethod) String() string {
+func (m TimeAggregationMethod) String() string {
 	b, err := m.MarshalText()
 	if err != nil {
 		return "%(INVALID)!"
 	}
 	return string(b)
 }
 
-func (m AggregateMethod) MarshalText() ([]byte, error) {
+func (m TimeAggregationMethod) MarshalText() ([]byte, error) {
 	switch m {
-	case AggregateDefault:
+	case TimeAggregationDefault:
 		return nil, nil
-	case AggregateCount:
+	case TimeAggregationCount:
 		return []byte("count"), nil
-	case AggregateMin:
+	case TimeAggregationMin:
 		return []byte("min"), nil
-	case AggregateMax:
+	case TimeAggregationMax:
 		return []byte("max"), nil
-	case AggregateSum:
+	case TimeAggregationSum:
 		return []byte("sum"), nil
-	case AggregateAvg:
+	case TimeAggregationAvg:
 		return []byte("avg"), nil
-	case AggregateStateHistSeconds:
+	case TimeAggregationSeconds:
 		return []byte("state-histogram-seconds"), nil
-	case AggregateStateHistPercent:
+	case TimeAggregationPercent:
 		return []byte("state-histogram-percent"), nil
-	case AggregateStateHistRate:
+	case TimeAggregationRate:
 		return []byte("state-histogram-rate"), nil
```

Comment:
Update to remove `-histogram` in result. The old values work, but the new ones are more canonical.